### PR TITLE
Eos noon master cherry pick

### PR DESCRIPTION
**Pull request by heifner**
Catching up eos-noon to master.
Does not include:
* tests/*
* scripts/*
* programs/launcher/*
* plugins/net*plugin/*
* .gitignore

### CODE REVIEW

**Comment by heifner** [comment]
Not ready for review...

**Comment by heifner** [comment]
Now ready for review.

File: Docker/eosc.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # Go into cmd loop: sudo ./eosc.sh
```
Comment (wanderingbort) [inline Docker/eosc.sh]: we should probably rename this and all of its in docker dependencies to `eosioc` to match the binaries

File: README.md [hunk 2]
```diff
@@ -258,15 +258,15 @@ Run the `create` command where `inita` is the account authorizing the creation o
 ./eosc create account inita currency PUBLIC_KEY_1 PUBLIC_KEY_2
```
Comment (wanderingbort) [inline README.md]: need to rename `eos*` binaries to `eosio*` in the docs

File: plugins/txn_test_gen_plugin/txn_test_gen_plugin.cpp [hunk 3]
```diff
@@ -127,14 +127,14 @@ struct txn_test_gen_plugin_impl {
          if(ec)
             return;
          try {
-            send_transaction();
+            this->send_transaction();
```
Comment (wanderingbort) [inline plugins/txn_test_gen_plugin/txn_test_gen_plugin.cpp]: we are pretty sure that the situation causing the need for the explicit `this->` was due to the use of `auto ec` at the time this change was made to master.  Can we reconfirm that this is necessary now that we are explicitly typing the input parameter?  

refer to: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=67274

File: Docker/eosc.sh (hunk 1 above)
Comment (heifner) [reply Docker/eosc.sh]: I updated the Docker/* with eosio. I will have our docker guy test and fix as needed.

File: plugins/txn_test_gen_plugin/txn_test_gen_plugin.cpp (hunk 3 above)
Comment (heifner) [reply plugins/txn_test_gen_plugin/txn_test_gen_plugin.cpp]: Ok. I removed this.

**Comment by heifner** [comment]
Updates ready for review.

**Pull request merged**